File: TA1S2016_Java/src/com/banco/Funcionario.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package com.banco;

import java.util.Date;

/**
 *
 * @author Antonio Lopes
 */
public class Funcionario {
   
    protected String nome;
    private String sobrenome;
    protected String cpf;
    protected Date dataNascimento;
    protected Double salario;

    public Funcionario(String nome, Double salario) {
        this.nome = nome;
        this.salario = salario;
    }
    
    public Double getBonificacao() {
        return this.salario * 0.10;
    }
    
    public String getNome() {
        return nome;
    }
    public void setNome(String nome) {
        this.nome = nome;
    }
    public void setNome(String nome, String sobrenome) {
        this.setNome(nome);
        this.sobrenome = sobrenome;
    }
    
    public String getSobrenome() {
        return sobrenome;
    }
    public void setSobrenome(String sobrenome) {
        this.sobrenome = sobrenome;
    }
    
    public String getCpf() {
        return cpf;
    }
    public void setCpf(String cpf) {
        this.cpf = cpf;
    }

    public Date getDataNascimento() {
        return dataNascimento;
    }
    public void setDataNascimento(Date dataNascimento) {
        this.dataNascimento = dataNascimento;
    }

    public Double getSalario() {
        return salario;
    }
    public void setSalario(Double salario) {
        this.salario = salario;
    }
}

File: TA1S2016_CSharp/TA1S2016_CSharp/Funcionario.cs
using System;

namespace TA1S2016_CSharp
{
    public class Funcionario
    {
        string nome;
        string sobrenome;
        string cpf;
        DateTime  dataNascimento;
        double salario;

        public virtual double Bonificacao
        {
            get { return this.salario * 0.10; }
        }

        public string Nome
        {
            get { return nome; }
            set { nome = value; }
        }

        public string Sobrenome
        {
            get { return sobrenome; }
            set { sobrenome = value; }
        }

        public string Cpf
        {
            get { return cpf; }
            set { cpf = value; }
        }

        public DateTime DataNascimento
        {
            get { return dataNascimento; }
            set { dataNascimento = value; }
        }

        public double Salario
        {
            get { return salario; }
            set { salario = value; }
        }
    }
}
